account_fiscal_position_vies_warning/models/res_partner.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2025 Moduon Team S.L. <info@moduon.team>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.constrains("property_account_position_id")
+    def check_fiscal_position_and_vies_valid(self):
+        if not self.env.company.vat_check_vies:
+            return
+        for record in self:
+            if (
+                record.property_account_position_id
+                in self.env.company.check_vies_fiscal_position_ids
+                and not record.vies_valid
+            ):
+                self.raise_vies_warning(
+                    record.name, record.property_account_position_id.name
+                )
+
+    def raise_vies_warning(self, partner_name, fiscal_position_name):
+        raise ValidationError(
+            _(
+                "You can't set the Fiscal Position to %(fiscal_pos)s, "
+                "%(partner_name)s doesn't pass VIES VAT number validation!"
+            )
+            % {
+                "fiscal_pos": fiscal_position_name,
+                "partner_name": partner_name,
```

Comment:
You can use `self` and take the data from the record:
```suggestion
    def raise_vies_warning(self, fiscal_position_id):
        raise ValidationError(
            _(
                "You can't set the Fiscal Position to %(fiscal_pos)s, "
                "%(partner_name)s doesn't pass VIES VAT number validation!"
            )
            % {
                "fiscal_pos": fiscal_position_id.name,
                "partner_name": self.name,
```